File: src/controller/controller_vilao_veiculo.py
from conexions.conexion import Conexion


class Controller_Vilao_Veiculo:
    def __init__(self):
        self.conn = Conexion('conexions/info/login.json', can_write=True)

    def inserir_vilao_veiculo(self):
        self.conn.connect()

        id_vilao = input("ID do Vilão: ")
        id_veiculo = input("ID do Veículo: ")

        if not self.verifica_existencia_vilao_veiculo(id_vilao, id_veiculo):
            self.conn.execute_non_query(f"""
                INSERT INTO Vilao_Veiculo (ID_Vilao, ID_Veiculo) 
                VALUES (:1, :2)
            """, (id_vilao, id_veiculo))
            print(f"Associação Vilão {id_vilao} com Veículo {id_veiculo} inserida com sucesso.")
        else:
            print(f"A associação Vilão {id_vilao} com Veículo {id_veiculo} já existe.")

    def excluir_vilao_veiculo(self):
        self.conn.connect()

        id_vilao = input("ID do Vilão: ")
        id_veiculo = input("ID do Veículo: ")

        if self.verifica_existencia_vilao_veiculo(id_vilao, id_veiculo):
            self.conn.execute_non_query(f"DELETE FROM Vilao_Veiculo WHERE ID_Vilao = {id_vilao} AND ID_Veiculo = {id_veiculo}")
            print(f"Associação Vilão {id_vilao} com Veículo {id_veiculo} removida com sucesso.")
        else:
            print(f"A associação Vilão {id_vilao} com Veículo {id_veiculo} não existe.")

    def verifica_existencia_vilao_veiculo(self, id_vilao: str, id_veiculo: str) -> bool:
        df_vilao_veiculo = self.conn.sql_to_data_frame(f"SELECT * FROM Vilao_Veiculo WHERE ID_Vilao = {id_vilao} AND ID_Veiculo = {id_veiculo}")
        return not df_vilao_veiculo.empty

File: src/controller/controller_vilao_veiculo.rb
require_relative '../conexions/connection'
require_relative '../model/vilao_veiculo'

class Controller_vilao_veiculo
  def initialize
    config_file = 'conexions/info/login.json'
    @connection = Connection.new(config_file)
  end

  def inserir_vilao_veiculo
    database = @connection.connect
    collection = database.database[:vilao_veiculo]

    puts "ID do Vilão:"
    id_vilao = gets.chomp
    puts "ID do Veículo:"
    id_veiculo = gets.chomp

    unless verifica_existencia_vilao_veiculo(id_vilao, id_veiculo)
      collection.insert_one({
                              "vilao_id" => id_vilao,
                              "veiculo_id" => id_veiculo
                            })
      puts "Associação Vilão #{id_vilao} com Veículo #{id_veiculo} inserida com sucesso."
    else
      puts "A associação Vilão #{id_vilao} com Veículo #{id_veiculo} já existe."
    end
  end

  def excluir_vilao_veiculo
    database = @connection.connect
    collection = database.database[:vilao_veiculo]

    puts "ID do Vilão:"
    id_vilao = gets.chomp
    puts "ID do Veículo:"
    id_veiculo = gets.chomp

    if verifica_existencia_vilao_veiculo(id_vilao, id_veiculo)
      collection.delete_one({
                              "vilao_id" => id_vilao,
                              "veiculo_id" => id_veiculo
                            })
      puts "Associação Vilão #{id_vilao} com Veículo #{id_veiculo} removida com sucesso."
    else
      puts "A associação Vilão #{id_vilao} com Veículo #{id_veiculo} não existe."
    end
  end

  def verifica_existencia_vilao_veiculo(id_vilao, id_veiculo)
    database = @connection.connect
    collection = database.database[:vilao_veiculo]
    result = collection.find({ "vilao_id" => id_vilao, "veiculo_id" => id_veiculo }).to_a
    !result.empty?
  end
end